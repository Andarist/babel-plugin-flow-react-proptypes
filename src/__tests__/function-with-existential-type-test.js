// http://sitr.us/2015/05/31/advanced-features-in-flow.html#existential-types
var babel = require('babel-core');
var content = `
type FooProps = {
  bar: *
}

const C = (props: FooProps) : ReactElement => {
}
`;

it('function-with-type-annotation', () => {
  var res = babel.transform(content, {
    babelrc: false,
    presets: ['es2015', 'stage-1', 'react'],
    plugins: ['syntax-flow', require('../')],
  }).code;
  expect(res).toMatchSnapshot();
});
