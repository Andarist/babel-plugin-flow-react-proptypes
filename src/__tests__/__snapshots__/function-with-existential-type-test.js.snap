// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`function-with-type-annotation 1`] = `
"\\"use strict\\";

var _propTypes = require(\\"prop-types\\");

var _propTypes2 = _interopRequireDefault(_propTypes);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

C.propTypes = {
  bar: function bar(props, propName, componentName) {
    if (!Object.prototype.hasOwnProperty.call(props, propName)) {
      throw new Error(\\"Prop \`\\" + propName + \\"\` has type 'any' or 'mixed', but was not provided to \`\\" + componentName + \\"\`. Pass undefined or any other value.\\");
    }
  }
};


var C = function C(props) {};"
`;
